File: spikes/telegram-engine/adapters/teleproto/smoke.mjs
/**
 * Explicit live smoke test for a dedicated Telegram test account.
 * It only connects, checks authorization, disconnects, and never prints secrets.
 */

import { SessionConfig, createTeleprotoAdapter } from "./adapter.mjs";

function required(name) {
  const value = process.env[name]?.trim();
  if (!value) throw new Error(`${name} wajib diisi untuk smoke test akun uji.`);
  return value;
}

try {
  const config = new SessionConfig({
    apiId: Number(required("TELEGRAM_TEST_API_ID")),
    apiHash: required("TELEGRAM_TEST_API_HASH"),
    session: required("TELEGRAM_TEST_SESSION"),
  });
  const adapter = createTeleprotoAdapter(config);
  await adapter.connect();
  const result = { scenario: "connect_authorized_disconnect", passed: true, ...adapter.describe() };
  await adapter.disconnect();
  process.stdout.write(`${JSON.stringify(result)}\n`);
} catch (error) {
  const code = error?.code ?? "SMOKE_SETUP_OR_CONNECT_FAILED";
  process.stdout.write(`${JSON.stringify({ scenario: "connect_authorized_disconnect", passed: false, code })}\n`);
  process.exitCode = 1;
}

File: spikes/telegram-engine/adapters/telethon/smoke.py
"""Explicit live connectivity smoke test for a dedicated Telegram test account.

This command never performs OTP login, sends no message, and prints no session/API hash.
"""

from __future__ import annotations

import asyncio
import json
import os
import sys

from adapter import SessionConfig, create_telethon_adapter


def required(name: str) -> str:
    value = os.environ.get(name, "").strip()
    if not value:
        raise RuntimeError(f"{name} wajib diisi untuk smoke test akun uji.")
    return value


async def main() -> int:
    try:
        config = SessionConfig(
            api_id=int(required("TELEGRAM_TEST_API_ID")),
            api_hash=required("TELEGRAM_TEST_API_HASH"),
            session=required("TELEGRAM_TEST_SESSION"),
        )
        adapter = create_telethon_adapter(config)
        await adapter.connect()
        result = {"scenario": "connect_authorized_disconnect", "passed": True, **adapter.describe()}
        await adapter.disconnect()
        print(json.dumps(result, separators=(",", ":")))
        return 0
    except Exception as error:
        code = getattr(error, "code", "SMOKE_SETUP_OR_CONNECT_FAILED")
        print(json.dumps({"scenario": "connect_authorized_disconnect", "passed": False, "code": code}, separators=(",", ":")))
        return 1


if __name__ == "__main__":
    sys.exit(asyncio.run(main()))
